solver/src/main/java/DbConstraintSolverZ3InDocker.java

```diff
@@ -0,0 +1,149 @@
+import org.evomaster.client.java.sql.internal.constraint.DbTableConstraint;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.shaded.org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+/**
+ * A smt2 solver implementation using Z3 in a Docker container.
+ */
+public class DbConstraintSolverZ3InDocker implements DbConstraintSolver {
+
+    public static final String Z3_DOCKER_IMAGE = "ghcr.io/z3prover/z3:ubuntu-20.04-bare-z3-sha-ba8d8f0";
+
+    // The default entrypoint runs directly z3, we need to keep it running
+    public static final String ENTRYPOINT = "while :; do sleep 1000 ; done";
+
+    private final String containerPath = "/smt2-resources/";
+    private final GenericContainer<?>  z3Prover;
+    private final String tmpFolderPath;
+    private static final org.slf4j.Logger LOGGER = org.slf4j.LoggerFactory.getLogger(DbConstraintSolverZ3InDocker.class.getName());
```

Comment:
for consistency with the rest of the codebase, logger objects should be called `log`